runtime/tests/org.eclipse.e4.core.tests/src/org/eclipse/e4/core/internal/tests/di/extensions/ServiceSupplierTestCase.java

```diff
@@ -73,11 +73,11 @@ public static class TestDisabledBean {
 		@Inject
 		@Optional
 		@Service(filterExpression = "(component=disabled)")
-		TestService disabledService;
+		volatile TestService disabledService;
```

Comment:
if this Lists are used concurrently its not enough to use volatile, they need to be Concurrent Lists, or synchronized

Reply:
True. The volatile was an attempt to introduce a memory barrier to ensure that changed values become eventually visible, but, of course, that does not apply to content changes of the list.

Reply:
Since the field is injected, one cannot easily influence the type of list, i.e., ensure that it is thread-safe. But gladly OSGi handles that case:

> By far the easiest method is to use field injection of a list of services. If you make this field volatile then DS will inject a new list whenever the set of tracked services changes.
See https://enroute.osgi.org/FAQ/300-declarative-services.html#tracking-multiple-services

So from my understanding, `volatile` should be correct here.